src/vm/docs/mod.rs

```diff
@@ -519,36 +555,41 @@ and outputs a list of the same type with max_len += 1.",
 };
 
 const ASSERTS_MAX_LEN_API: SpecialAPI = SpecialAPI {
-    input_type: "buff|list, uint",
-    output_type: "(optional buff|list)",
-    signature: "(as-max-len? buffer u10)",
-    description: "The `as-max-len?` function takes a length N (must be a literal) and a buffer or list argument, which must be typed as a list
+    input_type: "buff|list|string-ascii|string-utf8, uint",
+    output_type: "(optional buff|list|string-ascii|string-utf8)",
+    signature: "(as-max-len? sequence u10)",
+    description: "The `as-max-len?` function takes a length N (must be a literal) and a sequence
+(buffer, list or string) argument, which must be typed as a list
 or buffer of length M and outputs that same list or buffer, but typed with max length N.
 
 This function returns an optional type with the resulting sequence. If the input sequence is less than
 or equal to the supplied max-len, it returns `(some <sequence>)`, otherwise it returns `none`.",
     example: "(as-max-len? (list 2 2 2) u3) ;; Returns (some (2 2 2))
-(as-max-len? (list 1 2 3) u2) ;; Returns none"
+(as-max-len? (list 1 2 3) u2) ;; Returns none
+(as-max-len? \"hello\" u10) ;; Returns (some \"hello\")
+(as-max-len? 0x010203 u10) ;; Returns (some 0x010203)"
 };
 
 const LEN_API: SpecialAPI = SpecialAPI {
-    input_type: "buff|list",
+    input_type: "buff|list|string-ascii|string-utf8",
     output_type: "uint",
-    signature: "(len buffer)",
-    description: "The `len` function returns the length of a given buffer or list.",
-    example: "(len \"blockstack\") ;; Returns u10
+    signature: "(len sequence)",
+    description: "The `len` function returns the length of a given buffer, list or string.",
+    example: "
+(len \"blockstack\") ;; Returns u10
 (len (list 1 2 3 4 5)) ;; Returns u5
+(len 0x010203) ;; Returns u3
 ",
 };
 
 const ELEMENT_AT_API: SpecialAPI = SpecialAPI {
-    input_type: "buff|list A, uint",
-    output_type: "(optional buff|A)",
+    input_type: "buff|list|string-ascii|string-utf8 A, uint",
+    output_type: "(optional (buff 1)|A|(string-ascii 1)|(string-utf8 1))",
     signature: "(element-at sequence index)",
     description:
         "The `element-at` function returns the element at `index` in the provided sequence.
 If `index` is greater than or equal to `(len sequence)`, this function returns `none`.
-For strings and buffers, this function will return 1-length strings or buffers.",
+For strings or buffers, this function will return 1-length strings or buffers.",
```

Comment:
Can you add an example here for `buff`?

Reply:
I think there already is one:
`(element-at 0xfb01 u1) ;; Returns (some 0x01)`